src/abstract_domains.rs

```diff
@@ -32,9 +112,93 @@ pub enum ExpressionDomain {
     /// that always panics.
     Bottom,
 
-    /// An expression that is a compile time constant value, such as a numeric literal or a
-    /// function.
+    /// An expression that is a compile time constant value, such as a numeric literal or a function.
     CompileTimeConstant(ConstantValue),
+
+    /// An expression that is either if_true or if_false, depending on the value of condition.
+    ConditionalExpression {
+        // A condition that results in a Boolean value
+        condition: Box<AbstractDomains>,
+        // The value of this expression if join_condition is true.
+        if_true: Box<ExpressionDomain>,
+        // The value of this expression if join_condition is false.
+        if_false: Box<ExpressionDomain>,
```

Comment:
Won't `if_false` always be the negation of `if_true`? And if so, might it make sense to have `condition_expression : Box<ExpressionDomain>` and `true_branch: bool` instead to save on space?

Reply:
No, a conditional expression is exactly what it is in C, so if_false is not the negation of if_true.
It seems the names are confusing. I'll rename if_true to consequent and if_false to alternate, if that will help.

Reply:
Got it, thanks. I don't feel strongly about the renaming.